sw/device/sca/ecc256_keygen_serial.c

```diff
@@ -144,6 +198,71 @@ static void p256_run_keygen(uint32_t mode, const uint32_t *seed,
   sca_set_trigger_low();
 }
 
+static void ecc256_ecdsa_secret_keygen_batch(const uint8_t *data,
+                                             size_t data_len) {
+  uint32_t num_traces = 0;
+  uint32_t out[kEcc256SeedNumWords];
+  uint32_t batch_digest[kEcc256SeedNumWords];
+  uint8_t dummy[kEcc256SeedNumBytes];
+  SS_CHECK(data_len == sizeof(num_traces));
+  num_traces = read_32(data);
+
+  if (num_traces > kNumBatchOpsMax) {
+    LOG_ERROR("Too many traces for one batch.");
+    return;
+  }
+
+  // zero the batch digest
+  for (uint32_t j = 0; j < kEcc256SeedNumWords; ++j) {
+    batch_digest[j] = 0;
+  }
+
+  for (uint32_t i = 0; i < num_traces; ++i) {
+    if (run_fixed) {
+      memcpy((uint8_t *)batch_seeds[i], (uint8_t *)ecc256_seed,
```

Comment:
The `uint8_t` casts are technically a strict aliasing violation (there's a pretty good explanation [here](https://gist.github.com/shafik/848ae25ee209f698763cffee272a58f8)). Essentially, it's not in general legal in C to cast between integers of different sizes, it's only legal to cast to `char` types. Compilers generally implement `uint8_t` as `unsigned char`, but they're not required to, which means that if they stopped doing that for whatever reason it might break compiler assumptions about how integers work, which pointers are still in use, etc.

However, if you change these to `char *` or `unsigned char *`, they will be legal and still compile just fine :slightly_smiling_face: For `memcpy` you shouldn't even need to cast, in fact.

Reply:
Ah okay, thanks for the explanation! I didn't know that. 
Sounds like a good thing to do, I will change it.